rustworkx-core/src/coloring.rs

```diff
@@ -90,6 +91,60 @@ where
     colors
 }
 
+/// Color edges of a graph using a greedy approach.
+///
+/// This function works by greedily coloring the line graph of the given graph.
+///
+/// The coloring problem is NP-hard and this is a heuristic algorithm
+/// which may not return an optimal solution.
+///
+/// Arguments:
+///
+/// * `graph` - The graph object to run the algorithm on
+///
+/// # Example
+/// ```rust
+///
+/// use petgraph::graph::Graph;
+/// use petgraph::graph::EdgeIndex;
+/// use petgraph::Undirected;
+/// use rustworkx_core::dictmap::*;
+/// use rustworkx_core::coloring::greedy_edge_color;
+///
+/// let g = Graph::<(), (), Undirected>::from_edges(&[(0, 1), (1, 2), (0, 2), (2, 3)]);
+/// let colors = greedy_edge_color(&g);
+/// let mut expected_colors = DictMap::new();
+/// expected_colors.insert(EdgeIndex::new(0), 2);
+/// expected_colors.insert(EdgeIndex::new(1), 0);
+/// expected_colors.insert(EdgeIndex::new(2), 1);
+/// expected_colors.insert(EdgeIndex::new(3), 2);
+/// assert_eq!(colors, expected_colors);
+/// ```
+///
+pub fn greedy_edge_color<G>(graph: G) -> DictMap<G::EdgeId, usize>
+where
+    G: EdgeCount + IntoNodeIdentifiers + IntoEdges,
+    G::EdgeId: Hash + Eq,
+{
+    let (new_graph, edge_to_node_map): (
+        petgraph::graph::UnGraph<(), ()>,
+        HashMap<G::EdgeId, NodeIndex>,
+    ) = line_graph(&graph, || (), || ());
+
+    let colors = greedy_node_color(&new_graph);
+
+    let mut edge_colors: DictMap<G::EdgeId, usize> = DictMap::with_capacity(graph.edge_count());
+
+    for edge in graph.edge_references() {
+        let edge_index = edge.id();
+        let node_index = edge_to_node_map.get(&edge_index).unwrap();
+        let edge_color = colors.get(node_index).unwrap();
+        edge_colors.insert(edge_index, *edge_color);
+    }
+
+    edge_colors
```

Comment:
Not that it matters from a performance perspective, but I think this could be rewritten as:

```suggestion
    graph
        .edge_references()
        .map(|edge| {
            let edge_index = edge.id();
            let node_index = edge_to_node_map.get(&edge_index).unwrap();
            let edge_color = colors.get(node_index).unwrap();
            (edge_index, *edge_color)
        })
        .collect()
```